API/ev3_sensor.c

```diff
@@ -245,34 +251,26 @@ void* ReadSensorData(int sensorPort)
 			return readNewDumbSensor(sensorPort);
 			// Lightsensor
 		case COL_REFLECT:
-			return readUartSensor(sensorPort);
```

Comment:
These repeated readUartSensor calls are removed to leave the only one call for all relates cases.